ui_testing/pages/testunit_page.py

```diff
@@ -249,3 +249,16 @@ def set_qualitative_value(self, value=''):
         qualitative_value = self.base_selenium.find_element_in_element(destination_element='general:input',
                                                                        source_element='test_unit:qualitative_value')
         qualitative_value.send_keys(value)
+
+    # if random is false, take the arguments that are not None only
+    # if random is true, take all the arguments, check if a value is given, otherwise generate random values
+    def update_testunit(self, number, name, iterations, category, random=True):
+        if not random:
```

Comment:
and in case of random = True?

Reply:
Check the previous comment, [https://github.com/modeso-open-source/1lims-automation/pull/111#discussion_r361932891](url)

Reply:
@SalmaaRagab This link returns 404

Reply:
@0xIslamTaha  In case random is true, take all the arguments, check if a value is given take it, otherwise generate random values (update all data)

Reply:
@0xIslamTaha Yes, because I'm not implementing the update testunit function, I just put a function with a suggested implementation.